src/services/books.service.ts

```diff
@@ -1,124 +1,47 @@
-import { Author, Genre } from "@prisma/client";
-import prisma from "../utils/prisma";
+import booksRepository from "../repositories/books.repository";
+import { CreateBookDto, UpdateBookDto } from "../dtos";
+
+// create book
+export const createBook = async (bookData: CreateBookDto) => {
+  const book = await booksRepository.createBook(bookData);
+  return book;
+};
 
 // view all books
 export const getAllBooks = async () => {
-  const books = await prisma.book.findMany({
-    include: {
-      authors: {
-        select: {
-          name: true,
-        },
-      },
-      genres: {
-        select: {
-          title: true,
-        },
-      },
-    },
-  });
-
+  const books = await booksRepository.getAllBooks();
   return books;
 };
 
 // view book by title
-// getBookbyId
-export const getBook = async (title: string) => {
-  const book = await prisma.book.findFirst({
-    where: {
-      title: title,
-    },
-  });
-
-  return book;
+export const getAllBooksByTitle = async (title: string) => {
+  const books = await booksRepository.getAllBooksByTitle(title);
+  return books;
```

Comment:
@MohammedZayed1 https://github.com/Eight-Horsepeople-of-the-Graduation/core/pull/55#discussion_r1589731719